### DISCUSSION TITLE
vipsheader for TIFF, display resolution in DPI instead of mm?

### DISCUSSION THREAD

**Discussion by jrochkind** [libvips/libvips#3418] [Ideas]
So, as far as I can tell, the TIFF format confusingly has built-in metadata for "x resolution" and "y resolution" that is a unitless number, with another metadata field `Resolution Unit`, which can be "inches" or "cm". https://www.awaresystems.be/imaging/tiff/tifftags/resolutionunit.html

i believe I have a tiff that has xres and yres set to `400`, and it's units set to `inches`.   That is, it is 400 DPI/PPI. (dots-per-inch, pixels-per-inch).  (I'm not sure if using DPI/PPI is global or just American, if other countries use metric here? It is an annoying non-metric holdout!)

`exiftool` says:

```
X Resolution                    : 400
Y Resolution                    : 400
Resolution Unit                 : inches
```

However, vipsheader seems to be set on converting from inches to... milimeters maybe? `vipsheader -a` says:

```
xres: 15.748
yres: 15.748
resolution-unit: in
vips-loader: tiffload
```

This is the same file. It's still reporting to me that the "resolution unit" is inches, but... the xres and yres are NOT in inches, and are NOT (I believe) the values actually in the TIFF metadata, I think vipsheader has converted them?

Those, I think, are  pixels-per-millimeter, it's converted the 400 DPI to dots-per-mm.  

Especially since vips itself often takes `--dpi` as an argument, I really want to know the DPI, which was what happened to be in the original metadata in the first place in this case too. 

Thoughts?

**Comment by jcupitt** [comment]
Hi @jrochkind,

Yes, libvips is pixels per mm everywhere internally, it then converts to and from the format resolution on read/write.

The `resolution-unit` tag says what unit to write to the file with, for formats which support more than one unit.

**Reply by jrochkind** [reply]
OK, thanks, good to know it's intentional. 

But... there's a `--dpi` command line argument for some operations, but not a `--dpmm`, so it's not actually everywhere! But ok, I can convert (and hope for no rounding errors -- in this case I have to round to integer to get the right amount, and not 400.4 or whatever), or use another tool.

**Reply by jcupitt** [reply]
I think there's usually a good reason for the `--dpi` options, and they are generally not directly related to image resolution. In svgload, for example, it's used to set the DPI option for the svg renderer.
